### PR TITLE
Allow `diff_method=None` to be supported by QJIT

### PR DESCRIPTION
**Pull request by mlxd**
**Context:** PennyLane natively supports setting the `diff_method=None` to avoid support for gradients. When set and running qjit, this currently fails due to a type error. This PR rectifies that by concerting the `None` argument to a string.

**Description of the Change:** As above.

**Benefits:** Enabling QJIT of PennyLane circuits where `diff_method=None`.

**Possible Drawbacks:** None.

**Related GitHub Issues:**

### CODE REVIEW

File: frontend/test/pytest/test_gradient.py [hunk 1]
```diff
@@ -156,6 +156,9 @@ def f(x: float):
         qml.RX(x, wires=0)
         return qml.expval(qml.PauliZ(wires=0))
 
+    # Ensure None allows forward-pass to succeed
+    assert np.allclose(qjit(f)(1.0), np.cos(1.0))
```
Comment (rmoyard) [inline frontend/test/pytest/test_gradient.py:160]: Good catch :+1:

File: frontend/catalyst/jax_primitives.py [hunk 2]
```diff
@@ -298,7 +298,10 @@ def _func_def_lowering(ctx, fn, call_jaxpr) -> str:
         # if supported and parameter-shift otherwise. Emulating the same behaviour
         # would require generating code to query the device.
         # For simplicity, Catalyst instead defaults to parameter-shift.
-        diff_method = fn.diff_method if fn.diff_method != "best" else "parameter-shift"
+        if not fn.diff_method:
```
Comment (rmoyard) [inline frontend/catalyst/jax_primitives.py:301]:
```suggestion
        if fn.diff_method is None:
```

**Review by rmoyard — COMMENTED** [review]
Thanks for the catch, just one suggestion :+1:

**Review by dime10 — APPROVED** [review]

File: frontend/catalyst/jax_primitives.py [hunk 3]
```diff
@@ -298,7 +298,10 @@ def _func_def_lowering(ctx, fn, call_jaxpr) -> str:
         # if supported and parameter-shift otherwise. Emulating the same behaviour
         # would require generating code to query the device.
         # For simplicity, Catalyst instead defaults to parameter-shift.
-        diff_method = fn.diff_method if fn.diff_method != "best" else "parameter-shift"
+        if fn.diff_method is None:
+            diff_method = "None"
+        else:
+            diff_method = "parameter-shift" if fn.diff_method == "best" else fn.diff_method
```
Comment (maliasadi) [inline frontend/catalyst/jax_primitives.py:304]: Would this be sufficient as `str(None) == 'None'` and `str(str("")) == str("")` with almost no cost?
```suggestion
        diff_method = "parameter-shift" if fn.diff_method == "best" else str(fn.diff_method)
```

**Review by maliasadi — APPROVED** [review]
:tada:

**Pull request merged**